src/core/sagas/products.ts

```diff
@@ -11,24 +13,53 @@ import { Tasks as productDetailsTasks } from './product-details';
 export namespace Tasks {
   export function* fetchProducts(flux: FluxCapacitor, action: Actions.FetchProducts) {
     try {
-      const request = yield effects.select(Requests.search, flux.config);
-      const res = yield effects.call([flux.clients.bridge, flux.clients.bridge.search], request);
-
-      if (res.redirect) {
-        yield effects.put(flux.actions.receiveRedirect(res.redirect));
+      const [products, navigations]: [Results, Store.Recommendations.Navigation[]] = yield effects.all([
+        effects.call(fetchProductsRequest, flux, action),
+        effects.call(fetchNavigations, flux, action)
+      ]);
+      if (products.redirect) {
+        yield effects.put(flux.actions.receiveRedirect(products.redirect));
       }
-
-      if (flux.config.search.redirectSingleResult && res.totalRecordCount === 1) {
-        yield effects.call(productDetailsTasks.receiveDetailsProduct, flux, res.records[0]);
+      if (flux.config.search.redirectSingleResult && products.totalRecordCount === 1) {
+        yield effects.call(productDetailsTasks.receiveDetailsProduct, flux, products.records[0]);
       } else {
-        flux.emit(Events.BEACON_SEARCH, res.id);
-        yield effects.put(<any>flux.actions.receiveProducts(res));
+        flux.emit(Events.BEACON_SEARCH, (<any>products).id);
+        if (navigations && !(navigations instanceof Error)) {
+          products.availableNavigation = RecommendationsAdapter.sortAndPinNavigations(products.availableNavigation,
+                                                                                     navigations, flux.config);
+          yield effects.put(<any>flux.actions.receiveRecommendationsNavigations(navigations));
+        }
+        yield effects.put(<any>flux.actions.receiveProducts(products));
         flux.saveState(utils.Routes.SEARCH);
       }
     } catch (e) {
       yield effects.put(<any>flux.actions.receiveProducts(e));
     }
   }
+  export function* fetchProductsRequest(flux: FluxCapacitor, action: Actions.FetchProducts) {
+    const request = yield effects.select(Requests.search, flux.config);
+    return yield effects.call([flux.clients.bridge, flux.clients.bridge.search], request);
+  }
+
+  export function* fetchNavigations(flux: FluxCapacitor, action: Actions.FetchProducts) {
+    try {
+      const iNav = flux.config.recommendations.iNav;
+      if (iNav.navigations.sort || iNav.refinements.sort) {
+        const recommendationsUrl = RecommendationsAdapter.buildUrl(flux.config.customerId, 'refinements', 'Popular');
+        // tslint:disable-next-line max-line-length
+        const recommendationsResponse = yield effects.call(utils.fetch, recommendationsUrl, RecommendationsAdapter.buildBody({
+          size: 10,
```

Comment:
Should we add this to configuration? Maybe that's a next step (not a part of this ticket's work)

Reply:
Yeah it should probably be in there, @effervescentia, thoughts?